### PR TITLE
Feature/colored coin get balance 

### PR DESCRIPTION
**Pull request by Yamaguchi**
This PR supports colored coin for the API `blockchain.scripthash.get_balance`

### CODE REVIEW

File: src/new_index/schema.rs [hunk 1]
```diff
@@ -142,6 +141,43 @@ impl ScriptStats {
...
+
+impl StatsKey {
+    pub fn from_color_id(color_id: &Option<ColorIdentifier>) -> Self {
+        match color_id {
+            Some(c) => {
+                let payload = &serialize(c);
+                let payload = *array_ref![payload, 1, 32];
+                StatsKey(c.token_type.clone() as u8, payload)
+            }
+            None => StatsKey(0, [0; 32]),
```
Comment (azuchi) [inline src/new_index/schema.rs:159]: In Tapyrus Core, `TokenType#none` is defined. How about defining a similar type in rust-tapyrus?
https://github.com/chaintope/tapyrus-core/blob/master/src/coloridentifier.h#L17

Or it is better to define 0, [0; 32] as a constant.

File: src/new_index/schema.rs [hunk 2]
```diff
@@ -1476,31 +1514,64 @@ struct ScriptCacheKey {
...
                 code: b'A',
                 scripthash: full_hash(scripthash),
+                color_id: color_id,
             },
             value: bincode::serialize(&(stats, blockhash)).unwrap(),
         }
     }
 
-    pub fn key(scripthash: &[u8]) -> Bytes {
+    pub fn filter(scripthash: &[u8]) -> Bytes {
```
Comment (azuchi) [inline src/new_index/schema.rs:1545]: It seems that it is just generating prefix data. Isn't it a better name to leave `key`?

File: src/new_index/schema.rs (hunk 1 above)
Comment (Yamaguchi) [reply src/new_index/schema.rs:159]: https://github.com/chaintope/rust-tapyrus/pull/28

Comment (Yamaguchi) [reply src/new_index/schema.rs]: Update rust-tapyrus version(0.4.4) 
- support TPC token type.
- implement Hash trait for ColorIdentifier

And replace Hash key type from StatsKey to ColorIdeitifer

**Pull request merged**